src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java

```diff
@@ -209,22 +246,31 @@ public String visit(final SqlGroupBy groupBy) throws AdapterException {
 
     @Override
     public String visit(final SqlFunctionAggregate function) throws AdapterException {
-        final List<String> argumentsSql = new ArrayList<>();
+        final StringBuilder builder = new StringBuilder();
+        final String functionNameInSourceSystem = this.dialect.getAggregateFunctionAliases()
+                .getOrDefault(function.getFunction(), function.getFunctionName());
+        builder.append(functionNameInSourceSystem);
+        builder.append("(");
+        final List<String> renderedArguments = new ArrayList<>();
         for (final SqlNode node : function.getArguments()) {
-            argumentsSql.add(node.accept(this));
+            renderedArguments.add(node.accept(this));
         }
-        if (function.getFunctionName().equalsIgnoreCase("count") && argumentsSql.isEmpty()) {
-            argumentsSql.add(SqlConstants.ASTERISK);
+        boolean countFunction = function.getFunctionName().equals("COUNT");
+        if (countFunction && renderedArguments.size() > 1) {
+            builder.append("(");
         }
```

Comment:
This must be after DISTINCT. Please add a test for that.

Reply:
Added a test